.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+# Find full documentation here https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
+name: CI
+
+on:
+  pull_request:
+
+  # Manual invocation.
+  workflow_dispatch:
+
+  push:
+    branches:
+      - main
```

Comment:
Is there a reason to only run on push to main? The [TeamCity build](https://teamcity.gutools.co.uk/buildConfiguration/EditorialTools_Tagmanager?buildTypeTab=buildTypeSettings) seems to be triggered whenever changes are pushed to any branch - the VCS trigger branch filter is +:*

Reply:
Thanks Rebecca, that's helpful to know. I also asked @akash1810 about this as I've noticed both approaches in different builds, as well as a bit of [discussion](https://github.com/guardian/editorial-preferences/pull/50/files#r1336047049) between @twrichards and @davidfurey about what the consensus is.

Akash mentioned that GHA is free in public repos, so in this specific case it would be fine to run on each commit. Though he also said that it's uncommon to need to build on each commit, so probably not necessary. Good to have clarity on this, I'll add something to the CI with GHA doc!